### ISSUE TITLE
Ranger on MacOS (iTerm, Alacritty)

### ISSUE THREAD

**Issue by alechp** [ChristianChiarulli/nvim#27]
## QUICKFIX UPDATE
If you need ranger and run into the error below, you can use https://github.com/francoiscabrol/ranger.vim in the interim

Turns out this is already available in the `vim-plug/plugins.vim`; just swap the comments so it looks like this:
![Screen Shot 2020-06-15 at 5 49 26 PM](https://user-images.githubusercontent.com/3058939/84719570-8f8e9a80-af30-11ea-9c2e-9a6ed877975e.png)

#### Screenshots
* Opening Ranger
> ![Screen Shot 2020-06-15 at 11 58 01 AM](https://user-images.githubusercontent.com/3058939/84695316-820cec80-aeff-11ea-8ee1-33239d871c35.png)

* Attempting to open file
> <img width="826" alt="Screen Shot 2020-06-15 at 3 41 31 PM" src="https://user-images.githubusercontent.com/3058939/84712973-c956a580-af1e-11ea-9442-b382b8f6ab9a.png">
<details>
<summary>Detailed error log</summary>
<code>
Error while loading plugin 'client'
~           Error while loading plugin 'hacks'
~           Error while loading plugin 'host'
~           Error while loading plugin 'service'
~           Error detected while processing /usr/local/Cellar/neovim/0.4.3/share/nvim/runtime/plugin/man.vim:
~           line    9:
~           E10: \ should be followed by /, ? or &
~           line   10:
~           E10: \ should be followed by /, ? or &
~           Error detected while processing function <SNR>8_LoadRemotePlugins[1]..<SNR>8_GetManifest[1]..<SNR>8_
~           GetManifestPath:
~           line    7:
~           E117: Unknown function: stdpath
~           --- Autocommands ---
~           Error detected while processing /usr/local/Cellar/neovim/0.4.3/share/nvim/runtime/plugin/shada.vim:
~           line    9:
~           E10: \ should be followed by /, ? or &
~           line   10:
~           E10: \ should be followed by /, ? or &
~           line   11:
~           E10: \ should be followed by /, ? or &
~           --- Autocommands ---
~           line   13:
~           E10: \ should be followed by /, ? or &
~           line   14:
~           E10: \ should be followed by /, ? or &
~           --- Autocommands ---
~           line   16:
~           E10: \ should be followed by /, ? or &
~           line   17:
~           E10: \ should be followed by /, ? or &
~           line   18:
~           E10: \ should be followed by /, ? or &
~           line   19:
~           E10: \ should be followed by /, ? or &
~           line   20:
~           E35: No previous regular expression
~           line   21:
~           E10: \ should be followed by /, ? or &
~           line   22:
~           E10: \ should be followed by /, ? or &
~           --- Autocommands ---
~           line   24:
~           E10: \ should be followed by /, ? or &
~           line   25:
~           E10: \ should be followed by /, ? or &
~           line   26:
~           E10: \ should be followed by /, ? or &
~           line   27:
~           E10: \ should be followed by /, ? or &
~           line   28:
~           E10: \ should be followed by /, ? or &
~           line   29:
~           E10: \ should be followed by /, ? or &
~           --- Autocommands ---
~           line   31:
~           E10: \ should be followed by /, ? or &
~           line   32:
</code>
</details>

* Ignoring error & opening file via Ranger
> ![Screen Shot 2020-06-15 at 11 57 43 AM](https://user-images.githubusercontent.com/3058939/84695381-99e47080-aeff-11ea-8270-ec0404faaf90.png)
* * This is achieved by hitting "Enter" a couple of times after the initial error
* * General functionality is broken (typing seemed to insert random characters in the buffer, switching between modes is intermittent, etc)

#### Screencast
https://www.loom.com/share/6e9ed4755420454292cccef892b17671

---
---

### Attempted Solutions (Failed)
* Removed `ueberzug` from `.config/ranger/rc.conf`
	* Ran `:RnvimrSync` after doing so
---
* Installed ueberzug
	* Installed XQuartz for Mac, per: https://github.com/seebye/ueberzug/issues/61

---
---

### Exploring (TBD)
* https://github.com/kevinhwang91/rnvimr/issues/11
> this seems to be an identical issue but author closed it; will update findings later
* https://github.com/neovim/neovim/issues/9960
* https://github.com/ipod825/vim-netranger
* https://github.com/francoiscabrol/ranger.vim

**Comment by ChristianChiarulli** [comment]
I'm pretty sure ueberzug doesn't work on Mac. I haven't run into this issue yet though I'll let you know if it happens to me. I use rnvimr pretty regularly at work on my mac.

**Comment by alechp** [comment]
> I haven't run into this issue yet though I'll let you know if it happens to me. I use rnvimr pretty regularly at work on my mac.

Good to know possible. Is your work mac using the same config as your linux? I cloned the nvim repo as-is (on iMac & my MBP - neither worked), so I'm not quite sure what's missing

**Comment by alechp** [comment]
Problem solved!

### TLDR
❌ `brew install ranger`
✅ `pip install ranger-fm`

### Context
* Python environment not setup properly (finally got setup with pyenv & sourced global deps properly)

* `brew install ranger` didn't work for me; switching to `pip install ranger-fm` solved the problem

* While I was switching from brew to pip, I didn't realize that `ranger` is a dead project from 2015 & should be installing `ranger-fm` instead

**Comment by Mathijs-Bakker** [comment]
> Problem solved!
> 
> ### TLDR
> ❌ `brew install ranger`
> ✅ `pip install ranger-fm`
> 
> ### Context
> * Python environment not setup properly (finally got setup with pyenv & sourced global deps properly)
> * `brew install ranger` didn't work for me; switching to `pip install ranger-fm` solved the problem
> * While I was switching from brew to pip, I didn't realize that `ranger` is a dead project from 2015 & should be installing `ranger-fm` instead

## Ranger:
Had exactly the same! XD
- ```brew uninstall ranger```
- ```pip3 install ranger-fm```
Did it for me (Python 3.8).

## Ueberzug on MacOS:
**Ueberzug** also works like a charm on MacOS.
Just do as described here: #https://github.com/seebye/ueberzug/issues/61
